src/main/java/com/uid2/client/PublisherUid2Client.java

```diff
@@ -62,16 +61,28 @@ public TokenGenerateResponse generateTokenResponse(TokenGenerateInput tokenGener
 
 
         try (Response response = client.newCall(request).execute()) {
+            final String responseString = getResponse(response);
+            
             if (!response.isSuccessful()) {
-                throw new Uid2Exception("Unexpected code " + response);
+                throw new Uid2Exception("Unexpected code " + responseString);
             }
-
-            String responseString = response.body() != null ? response.body().string() : "";
+            
             return publisherUid2Helper.createTokenGenerateResponse(responseString, envelope);
         } catch (IOException e) {
             throw new Uid2Exception("Error communicating with api endpoint", e);
         }
     }
+    
+    private static String getResponse(Response response) {
+        try {
+            if(response == null) {
+                return "empty response";
+            }
+            return response.body() != null ? response.body().string() : response.toString();
```

Comment:
i think it's better to return [response.toString(](url)) than "" here?

Reply:
it was the 

_Unexpected code Response{protocol=h2, code=400, message=, url=https://operator-integ.uidapi.com/v2/token/generate}_ 

like the APAC pub saw previously